erpnext/controllers/accounts_controller.py

```diff
@@ -1204,6 +1204,7 @@ def set_purchase_order_defaults(parent_doctype, parent_doctype_name, child_docna
 	child_item.description = item.description
 	child_item.schedule_date = trans_item.get('schedule_date') or p_doc.schedule_date
 	child_item.uom = trans_item.get("uom") or item.stock_uom
+	child_item.warehouse = p_doc.set_warehouse
```

Comment:
we'll have to do something similar to how the warehouse is set in `set_sales_order_defaults`, there's a high chance `set_warehouse` may not be set

Reply:
Good point, have edited the code accoridng to your advice. Thank you.